pkg/text/check_required_scopes.go

```diff
@@ -0,0 +1,29 @@
+package text
+
+import (
+	"errors"
+)
+
+var errMissingRequiredScope = errors.New("required scope missing")
+
+// RequiredScopeChecker creates case sensitive strict checker, validating that
+// provided scope matches one from required scopes list
+func RequiredScopeChecker(rs []string) func(string) error {
+	check := make(map[string]struct{}, len(rs))
+
+	for _, s := range rs {
```

Comment:
I think it might be more readable just to check if the scope is in the required scopes instead of making a map with empty objects and then checking if the key exists. Just my 2 cents.

Reply:
fair enough - the intention was to make this check more scalable, as execution complexity would grow with both nb of commits and nb of scopes to check (N*M where N is nb of required scopes and M being nb of commits to scan). I think hardly anyone will be using nb of required scopes high enough to make an impact on execution performance here, so I think leaving the for a loop as more readable here makes sense - adjusted.